contentgrid-appserver-autoconfigure/src/main/java/com/contentgrid/appserver/autoconfigure/events/ContentGridEventsAutoConfiguration.java

```diff
@@ -0,0 +1,32 @@
+package com.contentgrid.appserver.autoconfigure.events;
+
+import com.contentgrid.appserver.autoconfigure.query.engine.JOOQQueryEngineAutoConfiguration;
+import com.contentgrid.appserver.domain.events.DomainEventDispatcher;
+import com.contentgrid.appserver.domain.events.EntityFormatter;
+import com.contentgrid.appserver.events.RabbitMqEventHandlers;
+import org.springframework.beans.factory.ObjectProvider;
+import com.contentgrid.appserver.events.EventHandlerConfiguration;
+import com.contentgrid.appserver.query.engine.api.EventConsumer;
+import org.springframework.boot.autoconfigure.AutoConfiguration;
+import org.springframework.boot.autoconfigure.amqp.RabbitAutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+
+@AutoConfiguration(before = {
+        JOOQQueryEngineAutoConfiguration.class,
+}, after = {
+        RabbitAutoConfiguration.class,
+})
```

Comment:
This does not have any conditional annotations that depend on `RabbitAutoConfiguration` having run, so that dependency is not necessary.

And the `before` ordering should probably be an `after` ordering on `JOOQQueryEngineAutoConfiguration`, because that is the configuration that has a dependency on this. Inverting the ordering like this is a bit confusing

Reply:
> Without after = RabbitAutoConfiguration, the ObjectProvider will always be empty.

As far as I understand, the `ObjectProvider` will not be empty.

There are 2 distinct phases during configuration:

1. Gathering of all configurations (including autoconfigurations) and bean definitions
2. Instantiation of beans from their definitions

The autoconfiguration ordering *only* affects the order in which beans are created in phase 1. It does not affect the ordering in phase 2 at all.

> With after = JOOQ instead of before, it will have a duplicate bean with the noopEventHandlers.

I meant to *move* the ordering to the `JOOQQueryEngineAutoConfiguration` class.